src/flip.jl

```diff
@@ -122,7 +124,7 @@ function flip_chain(graph::BaseGraph,
     """
     steps_taken = 0
     first_scores = score_initial_partition(graph, partition, scores)
-    chain_scores = Array{Dict{String, Any}, 1}([first_scores])
+    chain_scores = ChainScoreData(scores, [first_scores])

```

Comment:
Hmmm I think this should perhaps be `ChainScoreData(deepcopy(scores), [first_scores])`.

Since arrays are passed by reference you could do something like this:
```
...
scores = [
      MyScore_1,
      MyScore_2
]

chain_scores = flip_chain(graph, ... )

push!(scores, MyScore_3]

save_scores(chain_scores)
```
Since arrays are passed by reference, you could actually change the contents of the `scores` array after running the chain and it would change ChainScoreData.scores . The following picture I think would make that clearer:

<img width="518" alt="Screen Shot 2020-07-18 at 9 28 44 AM" src="https://user-images.githubusercontent.com/18491804/87853630-b8ac9d00-c8d9-11ea-9393-f827ef21ead0.png">

I guess one would argue that you could change `ChainScoreData.scores` if you `push!(chain_scores.scores, "AnotherScore")` but that would be much more conscious from the user instead of innocuously changing the `scores` array i think. What do you think?

Reply:
Gotcha! Yeah, I think I should definitely change it to `deepcopy` instead, that feels much more hygienic. Thanks for catching!